extension/healthcheckextensionv2/internal/status/aggregator.go

```diff
@@ -0,0 +1,227 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package status // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextensionv2/internal/status"
+
+import (
+	"fmt"
+	"sync"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+// Extensions are treated as a pseudo pipeline and extsID is used as a map key
+var (
+	extsID    = component.NewID("extensions")
+	extsIDMap = map[component.ID]struct{}{extsID: {}}
+)
+
+const (
+	ScopeAll        = ""
```

Comment:
why `""` and not `"all"`?

Reply:
This is aligned with how the gRPC healthcheck works: https://github.com/grpc/grpc-go/blob/master/health/grpc_health_v1/health_grpc.pb.go#L47-L50. Also, more practically, I'd end up adding a lot of conditions checking for empty string, and replacing it with `all`. eg.

```go
        pipeline := r.URL.Query().Get("pipeline")
	st, ok := s.aggregator.AggregateStatus(pipeline, s.settings.Status.Detailed)
	if !ok {
		w.WriteHeader(http.StatusNotFound)
		return
	}
```

Would have to become:
```go
        pipeline := r.URL.Query().Get("pipeline")
        if pipeline == "" {
            pipeline = "all"
        }
	st, ok := s.aggregator.AggregateStatus(pipeline, s.settings.Status.Detailed)
	if !ok {
		w.WriteHeader(http.StatusNotFound)
		return
	}
```

This would be the case in each of the http handlers and RPCs.

Reply:
I added `Scope` and `Detail` types to the aggregator. It retains the usability of the string / bool types, but makes them easier to understand and less error prone. Here are some examples of how it looks now:

```go
pipeline := r.URL.Query().Get("pipeline")
st, ok := s.aggregator.AggregateStatus(
    status.Scope(pipeline),
    status.Detail(s.settings.Status.Detailed),
)
```

```go
st, ok := agg.AggregateStatus(status.ScopeAll, status.IncludeSubtrees)
```